internal/cache/file/downloader/job.go

```diff
@@ -124,6 +126,76 @@ func (job *Job) Cancel() {
 	defer job.mu.Unlock()
 }
 
+// subscribe adds subscriber for download job and returns channel which is
+// notified when the download is completed at least till the subscribed offset
+// or in case of failure.
```

Comment:
Please correct the comment, we also notify the user in case of  Job get cancelled.